src/integrate/totalvi/script.py

```diff
@@ -0,0 +1,181 @@
+from typing import Tuple
+
+import logging
+import mudata
+from anndata import AnnData  # For type hints
+from mudata import MuData  # For type hints
+import numpy as np
+from pandas import DataFrame
+import scvi
+from scipy.sparse import issparse
+from torch.cuda import is_available as cuda_is_available
+try:
+    from torch.backends.mps import is_available as mps_is_available
+except ModuleNotFoundError:
+    # Older pytorch versions
+    # MacOS GPUs
+    def mps_is_available():
+        return False
+    
+import os
+
+### VIASH START
+par = {
+    "query": "resources_test/scvi_tools/totalvi_pbmc_query_test.h5mu",
+    "reference": "resources_test/scvi_tools/totalvi_pbmc_ref_test.h5mu",
+    "query_modality": "rna",
+    "query_proteins_key": None,
+    "reference_modality": "rna",
+    "reference_proteins_key": None,
+    "force_retrain": False,
+    "input_layer": None,
+    "obs_batch": "sample_id",
+    "var_input": None,
+    "output": "foo.h5mu",
+    "obsm_output": "X_integrated_totalvi",
+    "reference_model_path": "totalvi_model_reference/",
+    "query_model_path": "totalvi_model_query/",
+    "max_epochs": 400,
+    "max_query_epochs": 200,
+    "weight_decay": 0.0
+}
+### VIASH END
+
+
+def _setup_logger():  # Is there a place where to put common code to not repeat it across components?
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+
+
+def align_proteins_names(adata_reference: AnnData, mdata_query: MuData, adata_query: AnnData, reference_proteins_key: str, query_proteins_key: str) -> AnnData:
+    """Make sure that proteins are located in the same .obsm slot in reference and query. Pad query proteins with zeros if they are absent"""
+    proteins_reference = adata_reference.obsm[reference_proteins_key]
+
+    # If query has no protein data, put matrix of zeros 
```

Comment:
I am not familiar with totalVI. What does putting a matrix of zeroes here allow us to do? Running totalVI without the protein modality?

Reply:
Yes, exactly. Query and reference should have the same input format. If a query doesn't have some proteins, they should be padded by zeros. Protein data will be imputed after running the model